core/src/main/java/hudson/util/DirScanner.java

```diff
@@ -98,6 +104,7 @@ public void scan(File dir, FileVisitor visitor) throws IOException {
         private final String includes, excludes;
 
         private boolean useDefaultExcludes = true;
+        private boolean followSymlinks = true;
```

Comment:
AIUI, when this is `true` the behavior should be identical to before this change, right? It seems a little inconsistent to change the behavior for `FilePath#tar` but not `FilePath#archive`, should we change that one as well? (I'm not sure which behavior would make more sense as a default for `FilePath#copyRecursiveTo`.)

Reply:
Actually the `zip` command follows the symlinks by default (you need to use a switch not to follow them), while the tar doesn't.